include/libdnf/module/module_package.hpp

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_PACKAGE_HPP
+#define LIBDNF_MODULE_MODULE_PACKAGE_HPP
+
+#include "libdnf/module/module_profile.hpp"
+
+#include <string>
+#include <vector>
+
+struct _ModulemdModuleStream;
+
+
+namespace libdnf::module {
+
+
+struct ModulePackageId {
+public:
+    ModulePackageId() = default;
+    explicit ModulePackageId(int id) : id(id) {}
+
+    bool operator==(const ModulePackageId & other) const noexcept { return id == other.id; }
+    bool operator!=(const ModulePackageId & other) const noexcept { return id != other.id; }
+
+    // Corresponds to solvable id
+    int id{0};
+};
+
+
+// Represents one modulemd document (uniquely described by name:stream:version:context:arch, but there can theoretically be more objects with the same NSVCA)
+// @replaces libdnf:module/ModulePackage.hpp:class:ModulePackage
+class ModulePackage {
```

Comment:
Just a suggestion but what about to rename `ModulePackage` to `ModuleStream`?

In my opinion it would make things less confusing, every time I see `ModulePackage` I have to think to figure out how does it map to the module metadata structure, it also even stores: `_ModulemdModuleStream * md_stream` as its data.

Reply:
I agree that all of the naming around modules is confusing, but I think that `ModuleStream` would be just as misleading, because it doesn't represent a module stream.

Reply:
Yea you are right, if I understand correctly it represents one: NSVCA, that is one of these things: https://github.com/fedora-modularity/libmodulemd/blob/main/yaml_specs/modulemd_stream_v2.yaml

So it represent the same thing `libmodulemd` calls a stream? At least that would be consistent.

Reply:
`ModulePackage` is not nice but `ModuleStream` would be even more confusing. I have heard some other options like `ModuleItem`.

Reply:
Erm, seeing this not really knowing what's what, `ModulePackage` is extremely misleading to me, I would think it's an RPM package from a module. Please change :slightly_smiling_face:

Reply:
I renamed it to `ModuleItem`, because it doesn't have the problem of being confused with module stream or RPM package.